src/app/shared/components/protected-component/protected.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ProfileService } from '../../../core/profile/profile.service';
+
+@Pipe({ name: 'checkLogin', pure: false })
+export class CheckLoginPipe implements PipeTransform {
+
+    constructor(private profileService: ProfileService) {}
+
+    transform(isLoggedIn: boolean, roles: Array<string> ) {
+        if (isLoggedIn === undefined || isLoggedIn === this.profileService.isLoggedIn()) {
```

Comment:
boolean arguments is evil, why do you need it ?
It confuses since you have service to check that user is logged in